public/app/plugins/panel/geomap/layers/data/photosLayer.tsx

```diff
@@ -0,0 +1,212 @@
+
+// Configuration options for Circle overlays
+export interface PhotoConfig {
+  kind: 'square' | 'circle' | 'anchored' | 'folio';
+  border: number; // Sets border width around images
+  shadow: boolean; // Renders drop shadow behind images
+  crop: boolean; // Crops images to fill shape
+  src?: string; // Image source field
+  radius: number; // Image radius
+  color: string; // Border color
+}
+
+const defaultOptions: PhotoConfig = {
+  kind: 'square',
+  border: 2,
+  shadow: true,
+  crop: true,
+  radius: 20,
+  color: 'rgb(200, 200, 200)',
+};
+
+export const PHOTOS_LAYER_ID = 'photos';
+
+// Used by default when nothing is configured
+export const defaultPhotosConfig: MapLayerOptions<PhotoConfig> = {
+  type: PHOTOS_LAYER_ID,
+  name: '', // will get replaced
+  config: defaultOptions,
+  location: {
+    mode: FrameGeometrySourceMode.Auto,
+  },
+  tooltip: true,
+};
+
+// TODO Find a way to use SVG scaled to behave like a png, currently using base64 conversion
+//const unknownImage = '../../../../../public/img/icons/unicons/question-circle.svg';
+const unknownImage =
+  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiBjbGFzcz0iY3NzLWV5eDRkbyI+PHBhdGggZD0iTTExLjI5LDE1LjI5YTEuNTgsMS41OCwwLDAsMC0uMTIuMTUuNzYuNzYsMCwwLDAtLjA5LjE4LjY0LjY0LDAsMCwwLS4wNi4xOCwxLjM2LDEuMzYsMCwwLDAsMCwuMi44NC44NCwwLDAsMCwuMDguMzguOS45LDAsMCwwLC41NC41NC45NC45NCwwLDAsMCwuNzYsMCwuOS45LDAsMCwwLC41NC0uNTRBMSwxLDAsMCwwLDEzLDE2YTEsMSwwLDAsMC0uMjktLjcxQTEsMSwwLDAsMCwxMS4yOSwxNS4yOVpNMTIsMkExMCwxMCwwLDEsMCwyMiwxMiwxMCwxMCwwLDAsMCwxMiwyWm0wLDE4YTgsOCwwLDEsMSw4LThBOCw4LDAsMCwxLDEyLDIwWk0xMiw3QTMsMywwLDAsMCw5LjQsOC41YTEsMSwwLDEsMCwxLjczLDFBMSwxLDAsMCwxLDEyLDlhMSwxLDAsMCwxLDAsMiwxLDEsMCwwLDAtMSwxdjFhMSwxLDAsMCwwLDIsMHYtLjE4QTMsMywwLDAsMCwxMiw3WiI+PC9wYXRoPjwvc3ZnPgo=';
+
+/**
+ * Map layer configuration for circle overlay
+ */
+export const photosLayer: MapLayerRegistryItem<PhotoConfig> = {
+  id: PHOTOS_LAYER_ID,
+  name: 'Photos',
+  description: 'Render photos at each data point',
+  isBaseMap: false,
+  showLocation: true,
+  hideOpacity: true,
+  state: PluginState.alpha,
+
+  /**
+   * Function that configures transformation and returns a transformer
+   * @param map
+   * @param options
+   * @param theme
+   */
+  create: async (map: Map, options: MapLayerOptions<PhotoConfig>, eventBus: EventBus, theme: GrafanaTheme2) => {
+    // Assert default values
+    const config = {
+      ...defaultOptions,
+      ...options?.config,
+    };
+
+    const location = await getLocationMatchers(options.location);
+    const source = new FrameVectorSource(location);
+    const vectorLayer = new VectorLayer({
+      source,
+    });
+
+    let images: string[] = [];
+
+    vectorLayer.setStyle((feature: FeatureLike) => {
+      let src = unknownImage;
+      if (images.length > 0) {
+        const idx = feature.get('rowIndex') as number;
+        src = images[idx] ?? unknownImage;
```

Comment:
Do we check to see if the src is valid? When I mess up the src for Alaska in gdev demo I see following (and no ? image to replace it)

![Screen Shot 2022-10-26 at 2 59 37 PM](https://user-images.githubusercontent.com/22381771/198146783-988e463e-b2c0-4205-bf4c-dd42de26a77b.png)

Reply:
Will add some basic check for now on the src string. Ideally we would apply a check at the getImage function.

Reply:
I couldn't think of a general way to validate based on string, so this is the approach I'm taking for now:
1. Maintain an image index load tracking array
2. When images load (onload function), add index to that array
3. When returning vector layers, if index is in array of loaded images, render a background style (with a blank pixel as image source) and photo layer; otherwise, render background style, an error image, and the photo layer (which will be blank if it doesn't load).

Reply:
This is how that behaves: 
![Nov-02-2022 09-55-06](https://user-images.githubusercontent.com/60050885/199552640-a00ba252-b4ea-4f64-87d7-6fa7d92b0621.gif)